In [ ]:
#Importando biblioteca
import numpy as np

## Gerando listas com alturas, pesos, IMCs e Sexo

In [ ]:
#Gerando dados de média 1.75 e desvio padrão 0.30, sendo 2000 pessoas em uma distribuição normalizada.
alturas = np.random.normal(loc=1.75, scale=0.3, size=2000)
alturas = alturas.tolist() #Arrays não permitem manipulação, por isso, precisamos transformar em list
#Usando os valores da lista Alturas, multiplicando por 50 e adicionando um número aleatório entre -5 e 5 para criar o dado de peso, apendado a lista Pesos
pesos = []
for i in alturas:
  a = i*50
  b = float(np.random.uniform(-5,5))
  c = a + b
  pesos.append(c)
# Criando lista com imc
imc = []
for i, j in zip(pesos,alturas):
  x = i/(j**2)
  x = round(x,2)
  imc.append(x)
# Criando lista com gênero
genero = []
for i in range(0,2000):
  x=np.random.uniform(0,1)
  if x < 0.5:
    genero.append("M")
  else:
    genero.append("F")
# Criando lista com pessoas cardiopatas (S), possivelmente cardiopatas(P) ou sem problemas cardíacos (N).
cardiaco = []
for i in range(0,2000):
  if imc[i] <= 30:
    cardiaco.append("N")
  elif imc[i] < 40:
    cardiaco.append("P")
  elif imc[i] >=40:
    cardiaco.append("S")

In [ ]:
# Salvando todas as listas em uma lista única (lista de listas)
all_lists = alturas, pesos, imc, genero, cardiaco

In [ ]:
import pandas as pd
# Transformando a lista de listas criada em um DataFrame
df = pd.DataFrame(all_lists, index=["Altura","Peso","IMC","Gênero","Cardíaco"])
df.head()

0          1          2          3          4           5     \
Altura     1.604583   1.801186   1.726782   1.561343   1.532794    2.701393   
Peso      83.054464  88.349691  89.792705  77.031939  79.843314  135.657733   
IMC           32.26      27.23      30.11       31.6      33.98       18.59   
Gênero            F          F          M          M          M           F   
Cardíaco          P          N          P          P          P           N   

               6          7          8          9     ...       1990  \
Altura     1.888861   1.930182   1.676392   1.260987  ...   1.480545   
Peso      94.527858  96.979957  84.539082  63.333969  ...  72.408408   
IMC           26.49      26.03      30.08      39.83  ...      33.03   
Gênero            M          M          M          M  ...          M   
Cardíaco          N          N          P          P  ...          P   

               1991       1992       1993       1994       1995       1996  \
Altura     1.889654   1.949406    1.55152   1.808941   1.725277    1.55599   
Peso      90.002298  93.260695  72.760426  94.694837  87.557722  74.657816   
IMC           25.21      24.54      30.23      28.94      29.42      30.84   
Gênero            M          F          F          F          M          M   
Cardíaco          N          N          P          N          N          P   

                1997       1998       1999  
Altura      2.075266   2.301322   1.356789  
Peso      100.556883  118.90793  68.747232  
IMC            23.35      22.45      37.34  
Gênero             F          F          F  
Cardíaco           N          N          P  

[5 rows x 2000 columns]

In [ ]:
# Transpondo linhas x colunas
df = pd.DataFrame(all_lists, index=["Altura","Peso","IMC","Gênero","Cardíaco"]).T
df

Altura        Peso    IMC Gênero Cardíaco
0     1.604583   83.054464  32.26      F        P
1     1.801186   88.349691  27.23      F        N
2     1.726782   89.792705  30.11      M        P
3     1.561343   77.031939   31.6      M        P
4     1.532794   79.843314  33.98      M        P
...        ...         ...    ...    ...      ...
1995  1.725277   87.557722  29.42      M        N
1996   1.55599   74.657816  30.84      M        P
1997  2.075266  100.556883  23.35      F        N
1998  2.301322   118.90793  22.45      F        N
1999  1.356789   68.747232  37.34      F        P

[2000 rows x 5 columns]

# Explorando medidas de tendência central e frequências

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Altura    2000 non-null   object
 1   Peso      2000 non-null   object
 2   IMC       2000 non-null   object
 3   Gênero    2000 non-null   object
 4   Cardíaco  2000 non-null   object
dtypes: object(5)
memory usage: 78.2+ KB


In [ ]:
# Transformando números importados como strings em números reais
df["Altura"] = pd.to_numeric(df["Altura"])
df["Peso"] = pd.to_numeric(df["Peso"])
df["IMC"] = pd.to_numeric(df["IMC"])

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Altura    2000 non-null   float64
 1   Peso      2000 non-null   float64
 2   IMC       2000 non-null   float64
 3   Gênero    2000 non-null   object 
 4   Cardíaco  2000 non-null   object 
dtypes: float64(3), object(2)
memory usage: 78.2+ KB


In [ ]:
# Exportando para um .csv
df.to_csv("df.csv")

In [ ]:
# EDA das variáveis float64
df.describe()

Altura         Peso          IMC
count  2000.000000  2000.000000  2000.000000
mean      1.742019    87.069470    29.636240
std       0.299110    15.220417     5.899484
min       0.465852    22.968296    18.310000
25%       1.534915    76.668602    25.510000
50%       1.746653    87.274985    28.640000
75%       1.951235    97.612793    32.470000
max       2.701393   138.926790   105.840000

In [ ]:
# EDA das variáveis categóricas
freq_gen = df["Gênero"].value_counts(normalize=True)*100
freq_card = df["Cardíaco"].value_counts(normalize=True)*100

In [ ]:
freq_card

Cardíaco
N    60.4
P    34.2
S     5.4
Name: proportion, dtype: float64

In [ ]:
freq_gen

Gênero
F    50.25
M    49.75
Name: proportion, dtype: float64

# Análise final
É uma população que apresenta distribuição gaussiana cujos dados descritivos são os que seguem:
* Amostra: 2000 pessoas;
* 50.25% Mulheres e 49.75% Homens;
* Altura média 1.74m;
* Peso médio 87.1kg;
* IMC médio 29.64 (categorizando Obesidade - WHO)
* 5.4% das pessoas apresentam problemas cardíacos;
* 34.2% das pessoas possuem tendência a ter problemas cardíacos;
* 60.4% das pessoas não apresentam tendência.